src/features/auth/model/useAuth.ts

```diff
@@ -0,0 +1,57 @@
+import { useNavigate } from 'react-router-dom';
+import { useStore } from '@shared/lib/useStore';
+import type {
+  RegistrationOptions,
+  AuthResponse,
+  LoginOptions,
+} from '../interfaces';
+import { registrationApi, loginApi } from '../api';
+import { setCookies } from './setCookies';
+
+interface AuthReturnType {
+  handleSubmitLogin: (values: LoginOptions) => Promise<void>;
+  handleSubmitRegistration: (values: RegistrationOptions) => Promise<void>;
+}
+
+export const useAuth = (): AuthReturnType => {
+  const navigate = useNavigate();
+  const { userStore } = useStore();
+
+  const handleResponse = (
+    response: AuthResponse,
+    refreshTokenExpires: number
+  ): void => {
+    userStore.setUserData(response.user);
+    setCookies(
+      response.accessToken,
+      response.refreshToken,
+      refreshTokenExpires
+    );
+    navigate('/expenses');
+  };
+
+  const handleSubmitLogin = async (values: LoginOptions): Promise<void> => {
```

Comment:
```suggestion
  const onLoginSubmit = async (values: LoginOptions): Promise<void> => {
```

Action should have last position